main.py

```diff
@@ -215,65 +228,101 @@ def handleIndividualChapter(chapter: str | int) -> None:
+
+        if chapters is None or chapters == []:
+            print("No chapters selected. Exiting...")
+            exit()
+
+        # Get the chapter number from the chapter title
+        for i in range(len(chapters)):
+            chapters[i] = str(re.search(r"(?<=Chapter )\w+", chapters[i]).group(0))
         
 
-# Handle each chapter
-for chapter in tqdm.tqdm(chapters, desc="Total progress", unit="chapter", leave=False):
-    print("-" * os.get_terminal_size().columns)
-    print(Fore.YELLOW + "[-] " + "Processing chapter " + chapter + "...")
-    handleIndividualChapter(chapter.strip())
-    print(Fore.GREEN + "[✓] " + "Processed chapter " + chapter + "!")
-
-if args.open and len(chapters) == 1 and not args.merge:
-    os.startfile(os.path.join(outputdir, f"Arc {arcnumber} Chapter {chapter} Summary.txt"))
-elif args.open and not args.merge:
-    os.startfile(outputdir)
+
+    # Sort the chapters in ascending order
+    chapters.sort()
+
+    print(Fore.YELLOW + "[-] " + "Handling chapter(s): " + ", ".join(chapters))
+
+    if args.merge:
+        # Remove the temp folder if it already exists
+        if os.path.exists(outputdir):
+            shutil.rmtree(outputdir)
+            os.mkdir(outputdir)
+            
+
+    # Handle each chapter
+    for chapter in tqdm.tqdm(chapters, desc="Total progress", unit="chapter", leave=False):
+        print("-" * os.get_terminal_size().columns)
+        print(Fore.YELLOW + "[-] " + "Processing chapter " + chapter + "...")
+        handleIndividualChapter(chapter.strip())
+        print(Fore.GREEN + "[✓] " + "Processed chapter " + chapter + "!")
+
+    if args.open and len(chapters) == 1 and not args.merge:
+        os.startfile(os.path.join(outputdir, f"Arc {arcnumber} Chapter {chapter} Summary.txt"))
+    elif args.open and not args.merge:
+        os.startfile(outputdir)
+
+# if args.custom_text is true:
+else:
+
+    if args.merge:
+        # Remove the temp folder if it already exists
+        if os.path.exists(outputdir):
+            shutil.rmtree(outputdir)
+            os.mkdir(outputdir)
+
+    print(Fore.YELLOW + "[-] " + "Summarizing...")
+    summary = summarize(text)
+    print(Fore.GREEN + "[✓] " + "Summarized!")
+
+    if args.verbose:
+        print(Fore.GREEN + 'Summary'.center(os.get_terminal_size().columns))
+        print(summary)
+        print(Fore.CYAN + f"Words: {len(summary.split())} | Characters: {len(summary)}".center(os.get_terminal_size().columns))
+        print("-" * os.get_terminal_size().columns)
+    
+    # Write to file
+    with open(os.path.join(outputdir, os.path.basename(args.input) + " Summary.txt"), "w+", encoding="utf-8") as file:
+        file.write(summary)
+        file.seek(0)
+        fart = file.read()
+        fart = fart.strip()
+        file.seek(0)
+        file.write(fart)
+        file.truncate()
+        file.close()
+    
+    if args.open:
+        os.startfile(os.path.join(outputdir, os.path.basename(args.input) + " Summary.txt"))
```

Comment:
there has to be a way to not put all of this under an if statement right? please tell me there is otherwise im going to cry.

like can we use an inverse if here? i dont think we can really slap all of this into a function but putting it all under an if statement feels so janky

Reply:
a function is def better than the massive if statement so thats a good place to start